flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -386,10 +386,20 @@ object GenerateUtils {
         generateNonNullLiteral(literalType, fieldTerm, literalType)
 
       case TIMESTAMP_WITH_LOCAL_TIME_ZONE =>
-        val millis = unixTimestampToLocalDateTime(literalValue.asInstanceOf[Long])
+        val fieldTerm = newName("timestampWithLocalZone")
+        val millis = unixTimestampToLocalDateTime(
+              literalValue.asInstanceOf[TimestampString].getMillisSinceEpoch)
             .atZone(ctx.tableConfig.getLocalTimeZone)
             .toInstant.toEpochMilli
-        generateNonNullLiteral(literalType, millis + "L", literalValue)
+        val nanoOfMillis = SqlDateTimeUtils.getNanoOfMillisSinceEpoch(
```

Comment:
Why not getting nano from the previous `Instant` you got

Reply:
the previous `Instant` only contains millisecond part